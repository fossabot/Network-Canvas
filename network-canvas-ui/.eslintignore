dist
node_modules
coverage
docs/index.html
config
bin
server
