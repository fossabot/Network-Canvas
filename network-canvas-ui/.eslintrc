{
    "extends": "airbnb",
    "env": {
        "browser": true,
        "commonjs": true,
        "es6": true,
        "node": true
    },
    "globals": {
      "__DEV__"      : false,
      "__TEST__"     : false,
      "__PROD__"     : false,
      "__COVERAGE__" : false
    },
    "parser": "babel-eslint",
    "parserOptions": {
        ecmaVersion: 7,
        "ecmaFeatures": {
            "jsx": true,
            "spread": true,
            "experimentalObjectRestSpread": true,
            "object-shorthand": ["error", "always"]
        },
        "sourceType": "module"
    },
    "plugins": ["react"],
    "rules": {
      "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
      "react/forbid-prop-types": "off",
      "react/no-array-index-key": "off"
    }
}
