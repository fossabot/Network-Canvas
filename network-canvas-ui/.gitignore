.DS_STORE
*.log

node_modules

dist
coverage
npm-debug.log
