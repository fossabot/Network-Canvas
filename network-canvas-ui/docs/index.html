<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <title>Network Canvas - Style Guide</title>
    <meta name="description" content="Network Canvas - Style Guide">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link href="https://fonts.googleapis.com/css?family=Quicksand" rel="stylesheet">
</head>

<body class="view-login">
    <div id="root"></div>
</body>

</html>
