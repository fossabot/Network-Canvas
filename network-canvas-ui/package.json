{
  "name": "network-canvas-ui",
  "version": "1.1.0",
  "description": "Styles and React components for the Network Canvas project",
  "main": "lib/components/index.js",
  "scripts": {
    "lint-js": "eslint src",
    "lint-sass": "sass-lint styles",
    "lint": "eslint bin build config server src tests",
    "test": "better-npm-run test",
    "build": "better-npm-run compile",
    "clean": "rimraf dist",
    "compile": "better-npm-run compile",
    "lint:fix": "npm run lint -- --fix",
    "start": "better-npm-run start:prod",
    "serve": "better-npm-run start",
    "dev": "better-npm-run dev",
    "test:dev": "npm run test -- --watch",
    "deploy": "better-npm-run deploy",
    "deploy:dev": "better-npm-run deploy:dev",
    "deploy:prod": "better-npm-run deploy:prod",
    "codecov": "cat coverage/*/lcov.info | codecov",
    "clean:publish": "rimraf lib",
    "prepublish": "npm run clean:publish && babel ./src --out-dir ./lib --source-maps --copy-files --presets env,react,stage-0 --plugins=add-module-exports,transform-class-properties"
  },
  "betterScripts": {
    "compile": {
      "command": "node bin/compile",
      "env": {
        "DEBUG": "app:*"
      }
    },
    "dev": {
      "command": "nodemon bin/dev-server --ignore dist --ignore coverage --ignore tests --ignore src",
      "env": {
        "NODE_ENV": "development",
        "DEBUG": "app:*"
      }
    },
    "deploy": {
      "command": "npm run lint && npm run clean && npm run compile",
      "env": {
        "DEBUG": "app:*"
      }
    },
    "deploy:dev": {
      "command": "npm run deploy",
      "env": {
        "NODE_ENV": "development",
        "DEBUG": "app:*"
      }
    },
    "deploy:prod": {
      "command": "npm run deploy",
      "env": {
        "NODE_ENV": "production",
        "DEBUG": "app:*"
      }
    },
    "prepublish": {
      "command": "babel ./src --out-dir ./lib --source-maps --presets env,react --plugins babel-plugin-add-module-exports"
    },
    "start": {
      "command": "node bin/dev-server",
      "env": {
        "DEBUG": "app:*"
      }
    },
    "start:prod": {
      "command": "node bin/dev-server",
      "env": {
        "DEBUG": "app:*",
        "NODE_ENV": "production"
      }
    },
    "test": {
      "command": "node ./node_modules/karma/bin/karma start config/karma.config",
      "env": {
        "NODE_ENV": "test",
        "DEBUG": "app:*"
      }
    }
  },
  "dependencies": {
    "better-npm-run": "0.0.13",
    "chai": "^3.4.1",
    "chai-as-promised": "^6.0.0",
    "chai-enzyme": "^0.6.1",
    "cheerio": "^0.22.0",
    "classnames": "^2.2.5",
    "codecov": "^1.0.1",
    "compression": "^1.6.2",
    "css-loader": "^0.26.0",
    "cssnano": "^3.7.4",
    "debug": "^2.2.0",
    "enzyme": "^2.0.0",
    "express": "^4.14.0",
    "extract-text-webpack-plugin": "^1.0.0",
    "file-loader": "^0.9.0",
    "fs-extra": "^1.0.0",
    "gridle": "^2.0.48",
    "html-webpack-plugin": "^2.22.0",
    "imports-loader": "^0.7.0",
    "ip": "^1.1.2",
    "json-loader": "^0.5.4",
    "karma": "^1.7.0",
    "karma-coverage": "^1.0.0",
    "karma-mocha": "^1.0.1",
    "karma-mocha-reporter": "^2.0.0",
    "karma-phantomjs-launcher": "^1.0.2",
    "karma-webpack-with-fast-source-maps": "^1.9.2",
    "lodash": "^4.17.4",
    "mocha": "^3.0.1",
    "node-sass": "^4.0.0",
    "nodemon": "^1.10.2",
    "phantomjs-prebuilt": "^2.1.12",
    "postcss-loader": "^1.1.0",
    "prop-types": "^15.5.6",
    "react": "^15.0.0",
    "react-addons-test-utils": "^15.0.0",
    "react-dom": "^15.0.0",
    "react-redux": "^5.0.1",
    "react-retina-image": "^2.0.4",
    "react-router": "^3.0.0",
    "react-router-redux": "^4.0.8",
    "react-svg-loader": "^1.1.1",
    "redbox-react": "^1.2.10",
    "redux": "^3.6.0",
    "redux-thunk": "^2.0.0",
    "resolve-url-loader": "^2.0.2",
    "rimraf": "^2.5.4",
    "sass-loader": "^4.0.0",
    "sinon": "^1.17.5",
    "sinon-chai": "^2.8.0",
    "style-loader": "^0.13.1",
    "url-loader": "^0.5.6",
    "webpack": "^1.12.14",
    "webpack-dev-middleware": "^1.6.1",
    "webpack-hot-middleware": "^2.12.2",
    "yargs": "^6.3.0"
  },
  "devDependencies": {
    "babel-cli": "^6.24.1",
    "babel-core": "^6.17.0",
    "babel-eslint": "^7.2.3",
    "babel-loader": "^6.2.5",
    "babel-plugin-add-module-exports": "^0.2.1",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-env": "^1.5.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.22.0",
    "eslint": "^3.19.0",
    "eslint-config-airbnb": "^15.0.1",
    "eslint-config-react-app": "^1.0.4",
    "eslint-config-standard": "^6.0.0",
    "eslint-config-standard-react": "^4.0.0",
    "eslint-loader": "^1.7.1",
    "eslint-plugin-babel": "^4.0.0",
    "eslint-plugin-flowtype": "^2.34.0",
    "eslint-plugin-import": "^2.3.0",
    "eslint-plugin-jsx-a11y": "^5.0.3",
    "eslint-plugin-promise": "^3.0.0",
    "eslint-plugin-react": "^7.0.1",
    "eslint-plugin-standard": "^2.0.0",
    "sass-lint": "^1.10.2"
  }
}
