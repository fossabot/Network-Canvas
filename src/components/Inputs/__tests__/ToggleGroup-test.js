/* eslint-env jest */

import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';

import { ContextInput } from 'network-canvas-ui';
import FieldComponents from '../fieldComponents';

const setup = (props) => {
  const mockProps = {
    label: 'Toggle group',
    input: {
      value: null,
      name: 'toggle_group'
    },
    options: ['1', '2', '3'],
    colors: ['neon-carrot', 'kiwi', 'sea-serpent'],
    meta: {
      invalid: true,
      error: 'an error message',
    },
    ...props,
  };

  const component = shallow((
    <FieldComponents.ToggleGroup store={createStore(() => {})} {...mockProps} />
  ));

  return component;
};

describe('<ToggleGroup />', () => {
  it('should render', () => {
    const subject = setup();

    expect(subject).toMatchSnapshot();
  });

  it('should render a toggle for each option', () => {
    const subject = setup();

    expect(subject.find('ContextInput').length).toBe(3);
  });
});
