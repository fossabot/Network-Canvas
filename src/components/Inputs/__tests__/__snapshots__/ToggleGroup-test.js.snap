exports[`<ToggleGroup /> should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div>
        Toggle group
    </div>
    <div>
        <ContextInput
            checked={false}
            color="neon-carrot"
            label="1"
            name="toggle_group"
            onCheck={[Function]} />
        <ContextInput
            checked={false}
            color="kiwi"
            label="2"
            name="toggle_group"
            onCheck={[Function]} />
        <ContextInput
            checked={false}
            color="sea-serpent"
            label="3"
            name="toggle_group"
            onCheck={[Function]} />
    </div>
    <div>
        an error message
    </div>
</div>,
  "nodes": Array [
    <div>
      <div>
            Toggle group
      </div>
      <div>
            <ContextInput
                  checked={false}
                  color="neon-carrot"
                  label="1"
                  name="toggle_group"
                  onCheck={[Function]} />
            <ContextInput
                  checked={false}
                  color="kiwi"
                  label="2"
                  name="toggle_group"
                  onCheck={[Function]} />
            <ContextInput
                  checked={false}
                  color="sea-serpent"
                  label="3"
                  name="toggle_group"
                  onCheck={[Function]} />
      </div>
      <div>
            an error message
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ToggleGroup
        colors={
                Array [
                        "neon-carrot",
                        "kiwi",
                        "sea-serpent",
                      ]
        }
        input={
                Object {
                        "name": "toggle_group",
                        "value": null,
                      }
        }
        label="Toggle group"
        meta={
                Object {
                        "error": "an error message",
                        "invalid": true,
                      }
        }
        options={
                Array [
                        "1",
                        "2",
                        "3",
                      ]
        }
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ToggleGroup {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onClickOption": [Function],
        "props": Object {
          "colors": Array [
            "neon-carrot",
            "kiwi",
            "sea-serpent",
          ],
          "input": Object {
            "name": "toggle_group",
            "value": null,
          },
          "label": "Toggle group",
          "meta": Object {
            "error": "an error message",
            "invalid": true,
          },
          "options": Array [
            "1",
            "2",
            "3",
          ],
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div>
                    Toggle group
          </div>
          <div>
                    <ContextInput
                              checked={false}
                              color="neon-carrot"
                              label="1"
                              name="toggle_group"
                              onCheck={[Function]} />
                    <ContextInput
                              checked={false}
                              color="kiwi"
                              label="2"
                              name="toggle_group"
                              onCheck={[Function]} />
                    <ContextInput
                              checked={false}
                              color="sea-serpent"
                              label="3"
                              name="toggle_group"
                              onCheck={[Function]} />
          </div>
          <div>
                    an error message
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div>
                    Toggle group
          </div>
          <div>
                    <ContextInput
                              checked={false}
                              color="neon-carrot"
                              label="1"
                              name="toggle_group"
                              onCheck={[Function]} />
                    <ContextInput
                              checked={false}
                              color="kiwi"
                              label="2"
                              name="toggle_group"
                              onCheck={[Function]} />
                    <ContextInput
                              checked={false}
                              color="sea-serpent"
                              label="3"
                              name="toggle_group"
                              onCheck={[Function]} />
          </div>
          <div>
                    an error message
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ToggleGroup
    colors={
        Array [
            "neon-carrot",
            "kiwi",
            "sea-serpent",
          ]
    }
    input={
        Object {
            "name": "toggle_group",
            "value": null,
          }
    }
    label="Toggle group"
    meta={
        Object {
            "error": "an error message",
            "invalid": true,
          }
    }
    options={
        Array [
            "1",
            "2",
            "3",
          ]
    }
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    } />,
}
`;
